### PR TITLE
Blockquote component

### PR DESCRIPTION
**Pull request by ogzcode**
#219

Hi,

1.I created the Blockquote component according to the code layout of the project. 
2.I made it according to the examples and style scheme on the Flowbite and React-Flowbite sites. 
3.I created the doc page for Blockquote and added it to the navigation.
4.I created sample components for the doc page and added their usage as well.
5.I did not add all the description content on the Flowbite site, I just created examples for size, align and type and added them to the doc page.

I hope the component turned out the way you wanted. I worked hard for this and I would be very happy if you accept my PR. 

I will be waiting for your feedback.

@Sqrcz @cogor @zoltanszogyenyi @robert1508

## Summary by CodeRabbit

- **New Features**
  - Introduced a comprehensive documentation file for blockquote components, enhancing usability and navigation.
  - Added several new Vue.js components demonstrating various styles and configurations of blockquotes, including alignment and size options.
  - Implemented a new customizable blockquote component to provide flexible usage in applications.

- **Bug Fixes**
  - No bug fixes were included in this release.

- **Documentation**
  - Enhanced documentation with detailed examples and usage instructions for blockquote components, facilitating easier integration for developers.

### CODE REVIEW

File: src/components/FwbBlockquote/FwbBlockquote.vue
Comment (Sqrcz) [inline src/components/FwbBlockquote/FwbBlockquote.vue:1]: as part of typography this file should be inside `src/components/Typography`

File: src/components/FwbBlockquote/composables/useBlockquoteClasses.ts
Comment (Sqrcz) [inline src/components/FwbBlockquote/composables/useBlockquoteClasses.ts:1]: it's fairly simple component... I don;t think we need dedicated composable for it.

take a look at `src/components/Typography/FwbP.vue` how it uses default classes and merges them with user classes using our dedicated composable `src/composables/useMergeClasses.ts`

File: src/components/FwbBlockquote/FwbBlockquote.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    type?: BlockquoteType;
+    size?: BlockquoteSize;
+    align?: BlockquoteAlign;
+    cite?: string;
+}
+
+const props = withDefaults(defineProps<BlockquoteProps>(), {
+    type: 'default',
+    size: 'md',
+    align: 'left',
```
Comment (Sqrcz) [inline src/components/FwbBlockquote/FwbBlockquote.vue:22]: size and alignment should be changed just by overriding tailwind classes... and not by dedicated props.

(again, take a look at `src/components/Typography/FwbP.vue` to see how it's done there)

File: src/components/FwbBlockquote/composables/useBlockquoteClasses.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const blockquoteAlignClasses: Record<BlockquoteAlign, string> = {
+  left: "text-left",
+  center: "text-center",
+  right: "text-right",
+};
+const blockquoteSizeClasses: Record<BlockquoteSize, string> = {
+  sm: "text-lg",
+  md: "text-xl",
+  lg: "text-2xl",
+};
```
Comment (Sqrcz) [inline src/components/FwbBlockquote/composables/useBlockquoteClasses.ts:15]: as stated above... this should be achieved by tailwind classes directly on the component

File: src/components/FwbBlockquote/composables/useBlockquoteClasses.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+) {
+  const attrs = useAttrs();
+  const blockquoteClasses = computed(() => {
+    return twMerge(
+      blockquoteSizeClasses[props.size],
+      blockquoteAlignClasses[props.align],
+      props.type === "solid" ? "bg-gray-100 dark:bg-gray-800 border-l-4 border-gray-300 p-4 dark:border-gray-500" : "",
+      defaultBlockquoteClasses,
+      attrs.class as string
+    );
```
Comment (Sqrcz) [inline src/components/FwbBlockquote/composables/useBlockquoteClasses.ts:34]: we have dedicated composable fot this `src/components/Typography/FwbP.vue` has a simple use case example

**Review by Sqrcz — CHANGES_REQUESTED** [review]
Hey, 

Thank you for this... It's always good to increase number of available components 🎉 

Please take a look at the comments below as well as the linter that is set up within a project... currently running `npm run lint` on this branch results in this:

```
✖ 85 problems (60 errors, 25 warnings)
  57 errors and 25 warnings potentially fixable with the `--fix` option.
```

Please fix those few things so we can maintain same code style across all over our components. 🙏 

Thanks 🙌

File: src/components/FwbBlockquote/FwbBlockquote.vue (hunk 1 above)
Comment (ogzcode) [reply src/components/FwbBlockquote/FwbBlockquote.vue:22]: I'm asking to be sure if what you mean here and in the messages below is this: the user should change states such as size and align by writing them in the class, not with props, and these props should be removed.
Can you please correct me if I misunderstood?

**Comment by ogzcode** [comment]
1. I put the FwbBlockquote component under typograpy.
2. I removed the props for size and alignment.
3. I edited the example files and updated the doc page as well.
4. I implemented the lint rules (finally)

I hope it turned out the way you wanted.

@Sqrcz

**Review by Sqrcz — APPROVED** [review]
Thank you for all of the changes. Now logic is way simpler and does the same thing 🎉 

There are some small lint issues in example files (you can see comments added by GH Action)... to mitigate that you could configure your code editor to "lint on save" or run `npm run lint-fix` after making changes. ;)

Let's not block this one though... Good work we have a new component :)

**Pull request merged**